### PR TITLE
Fix issue with /proc hidepid option

### LINKED ISSUE #4633: fakeroot option does not work with hidepid procfs
### Version of Singularity:

What version of Singularity are you using? Run:

```
$ singularity version
3.4.1
```

### Expected behavior

What did you expect to see when you do...?

```
$ singularity shell -f -w ubuntu/
Singularity ubuntu:~>
```

### Actual behavior

```
singularity shell -f -w ubuntu/
ERROR  : Failed to change directory to /proc/21020: No such file or directory
```

What actually happend? Why was it incorrect?

This error occurs when hidepid=1 or 2 option is used in the procfs with fakeroot option.
Is it possible to use fakeroot with hidpid?

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

Mount procfs with hidepid=1 or 2 and execute `singularity -f`

### What OS/distro are you running

```
$ cat /etc/os-release
NAME="SLES"
VERSION="12-SP2"
VERSION_ID="12.2"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP2"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp2"

```

### How did you install Singularity

Write here how you installed Singularity. Eg. RPM, source.

From the source

### PR DESCRIPTION
**Pull request by cclerget**
## Description of the Pull Request (PR):

Actually `hidepid=1` or `hidepid=2` prevent Singularity from working correctly. This PR make it works correctly while `hidepid=` is set, unfortunately there is one case where `hidepid` require to be disabled, it's possible to start an instance with setuid workflow but it will join during instance join and won't stop all processes correctly during instance stop, this PR handle this case and show an error explaining to disable `hidepid`.

### This fixes or addresses the following GitHub issues:

 - Fixes #4633

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: cmd/internal/cli/actions_linux.go [hunk 1]
```diff
@@ -94,6 +95,27 @@ func convertImage(filename string, unsquashfsPath string) (string, error) {
...
+func hidepidProc() bool {
+	entries, err := proc.GetMountInfoEntry("/proc/self/mountinfo")
+	if err != nil {
+		sylog.Warningf("while reading /proc/self/mountinfo: %s", err)
+		return false
+	}
+	for _, e := range entries {
+		if e.Point == "/proc" {
+			for _, o := range e.SuperOptions {
+				if strings.HasPrefix(o, "hidepid=") {
```
Comment (sashayakovtseva) [inline cmd/internal/cli/actions_linux.go:111]: What if hidepid=0?

Comment (cclerget) [reply cmd/internal/cli/actions_linux.go:111]: `hidepid` appears only if > 0

**Review by sashayakovtseva — APPROVED** [review]

**Comment by dtrudg** [comment]
I came across this today, in the Debian buster upgrade notes

https://www.debian.org/releases/stable/amd64/release-notes/ch-information.html

> 5.1.1. Hidepid mount option for procfs unsupported

> Using the hidepid mount option for /proc is known to cause problems with current versions of systemd, and is considered by systemd upstream to be an unsupported configuration. Users who have modified /etc/fstab to enable this option are advised to disable it before the upgrade, to ensure login sessions work on buster. (A possible route to re-enabling it is outlined on the wiki's Hardening page.) 

Given this is an unsupported systemd configuration, is it something we can anticipate seeing often?

**Comment by cclerget** [comment]
@dctrud I don't think that's something we will see often for now as it can break user space applications but we may need to run in hardened environments, additionally is systemd will always consider it as an unsupported configuration ? Actually the fix is not complete as instances started with setuid workflow can't be joined.

**Comment by dtrudg** [comment]
> additionally is systemd will always consider it as an unsupported configuration ?

It looks like it according to a comment from Lennart Poettering - but then lower down in that thread it looks like people may be working on things to get around this, which require kernel changes.

https://github.com/systemd/systemd/issues/12955#issuecomment-508490893
> But anyway, sorry. hidepid= the way it is is not compatible with systemd, and that's not going to change.

**Review by dtrudg — APPROVED** [review]

**Pull request merged**